From e6300998b5d521d161020cdd7296346c179db352 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Thu, 6 Jun 2019 14:59:32 +0800
Subject: [PATCH 6/8] ARM:anatop:fix compile error with out IMX_GPC

imx_gpc_usb_wakeup_enabled() & imx_gpc_enet_wakeup_enabled() defined when
enable IMX_GPC, use this marco to protect the code.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 arch/arm/mach-imx/anatop.c | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/anatop.c b/arch/arm/mach-imx/anatop.c
index e295d55b88c4..77f4efce8eed 100644
--- a/arch/arm/mach-imx/anatop.c
+++ b/arch/arm/mach-imx/anatop.c
@@ -150,8 +150,11 @@ void imx_anatop_pre_suspend(void)
 		imx_anatop_disable_pu(true);
 
 	if ((imx_mmdc_get_ddr_type() == IMX_DDR_TYPE_LPDDR2 ||
-		imx_mmdc_get_ddr_type() == IMX_MMDC_DDR_TYPE_LPDDR3) &&
-		!imx_gpc_usb_wakeup_enabled() && !imx_gpc_enet_wakeup_enabled())
+		imx_mmdc_get_ddr_type() == IMX_MMDC_DDR_TYPE_LPDDR3)
+#ifdef 	CONFIG_HAVE_IMX_GPC
+		&& !imx_gpc_usb_wakeup_enabled() && !imx_gpc_enet_wakeup_enabled()
+#endif
+		)
 		imx_anatop_enable_2p5_pulldown(true);
 	else
 		imx_anatop_enable_weak2p5(true);
@@ -180,8 +183,11 @@ void imx_anatop_post_resume(void)
 		imx_anatop_disable_pu(false);
 
 	if ((imx_mmdc_get_ddr_type() == IMX_DDR_TYPE_LPDDR2 ||
-		imx_mmdc_get_ddr_type() == IMX_MMDC_DDR_TYPE_LPDDR3) &&
-		!imx_gpc_usb_wakeup_enabled() && !imx_gpc_enet_wakeup_enabled())
+		imx_mmdc_get_ddr_type() == IMX_MMDC_DDR_TYPE_LPDDR3)
+#ifdef  CONFIG_HAVE_IMX_GPC
+		&& !imx_gpc_usb_wakeup_enabled() && !imx_gpc_enet_wakeup_enabled()
+#endif
+		)
 		imx_anatop_enable_2p5_pulldown(false);
 	else
 		imx_anatop_enable_weak2p5(false);
-- 
2.17.1

